src/gui/folderwatcher.cpp

```diff
@@ -190,4 +206,62 @@ void FolderWatcher::changeDetected(const QStringList &paths)
     }
 }
 
+void FolderWatcher::possiblyAddUnlockedOfficeFile(const QString &path, QSet<QString> &container)
+{
+    qCInfo(lcFolderWatcher) << "Checking if it is an office lock file:" << path;
+    QString officeLockFilePrefixFound;
+    for (const auto &currentPrefix : officeLockPrefixes) {
+        if (path.contains(currentPrefix)) {
+            officeLockFilePrefixFound = currentPrefix;
+            break;
+        }
+    }
+
+    if (officeLockFilePrefixFound.isEmpty() || QFileInfo::exists(path)) {
+        return;
+    }
+
+    qCInfo(lcFolderWatcher) << "Found an office lock file with prefix:" << officeLockFilePrefixFound << "in path:" << path;
+
+    const auto lockFilePathWitoutPrefix = QString(path).replace(officeLockFilePrefixFound, QStringLiteral(""));
+    auto lockFilePathWitoutPrefixSplit = lockFilePathWitoutPrefix.split(QLatin1Char('.'));
+
+    if (lockFilePathWitoutPrefixSplit.size() < 2) {
+        return;
+    }
+
+    const auto lockFileExtensionDirty = lockFilePathWitoutPrefixSplit.takeLast();
+    // remove possible non-alphabetical characters at the end of the extension
+    QString extensionSanitized;
+    for (const auto &ch : lockFileExtensionDirty) {
+        if (ch.isLetter()) {
+            extensionSanitized += ch;
+        }
+    }
+    lockFilePathWitoutPrefixSplit.push_back(extensionSanitized);
+    auto unlockedFilePath = lockFilePathWitoutPrefixSplit.join(QLatin1Char('.'));
+
+    if (!QFile::exists(unlockedFilePath)) {
+        qCInfo(lcFolderWatcher) << "Assumed unlocked file path" << unlockedFilePath << "does not exist.Going to try to find matching file";
+        // we are having a different case now (Office 365 modifies the name by removing 2 first symbols)
+        auto splitFilePath = unlockedFilePath.split(QLatin1Char('/'));
+        unlockedFilePath.clear();
+        if (splitFilePath.size() > 1) {
+            const auto lockFileName = splitFilePath.takeLast();
+            const QDir dir(splitFilePath.join(QLatin1Char('/')));
+            for (const auto &candidateUnlockedFileInfo : dir.entryInfoList(QDir::Files)) {
+                if (candidateUnlockedFileInfo.fileName().contains(lockFileName)) {
+                    unlockedFilePath = candidateUnlockedFileInfo.absoluteFilePath();
+                    break;
+                }
+            }
+        }
+    }
```

Comment:
same comment
would be better to split that into an utility function
that would allow to have different quirky methods to solve integration issue with specific software (think about a newer version of the office software that you have tested but with yet another weird behavior